### PR TITLE
update `xcode` to not use attributes, only auth dev account if out-of-date

### PR DESCRIPTION
**Pull request by jazaval**
# Changelog

### Fixed

- Extracted authentication with Apple via `xcode` resource away `xcode` object instantiation, resolving #234. 

### Added

- Added `apple_id` property to `xcode` resource to remove dependency on attributes or data bags for authentication.

### Removed

- Removed the ability to authenticate with Apple for `xcode` downloads via node attributes or data bags.
- Removed last default cookbook attribute:`node['macos']['apple_id']`

### CODE REVIEW

File: libraries/xcode.rb [hunk 1]
```diff
@@ -3,18 +3,20 @@
...
-    attr_reader :download_url
+    attr_reader :version, :intended_path, :download_url
 
     def initialize(semantic_version, intended_path, download_url = '')
       @semantic_version = semantic_version
       @intended_path = intended_path
       @download_url = download_url
-      @version = if download_url.empty?
-                   latest_xcode_revision(Xcode::Version.new(semantic_version))
+      @version = -> { determine_version }.call
```
Comment (jweyer8) [inline libraries/xcode.rb:12]: Possible alternative to the lamda:

`@version = download_url.empty? ? latest_xcode_revision(Xcode::Version.new(@semantic_version)) : @samantic_version`

**Review by jweyer8 — COMMENTED** [review]
Looks good!

**Review by jweyer8 — APPROVED** [review]

File: libraries/xcode.rb (hunk 1 above)
Comment (jazaval) [reply libraries/xcode.rb]: The lambda allows the `xcode` object to be instantiated without authenticating with Apple to see what versions are available, and allows us to first check things like `xcode.installed?` for the resource guards. The ternary would still be evaluated at initialization, which will require auth with Apple right away.

**Review by mjmerin — APPROVED** [review]
LGTM

**Pull request merged**